packages/excalidraw/utils.ts

```diff
@@ -671,6 +671,18 @@ export const arrayToMapWithIndex = <T extends { id: string }>(
     return acc;
   }, new Map<string, [element: T, index: number]>());
 
+/**
+ * Transform array into an object, use only when array order is irrelevant.
+ */
+export const arrayToObject = <T>(
+  array: readonly T[],
+  groupBy?: (value: T) => string,
+) =>
+  array.reduce((acc, value) => {
+    acc[groupBy ? groupBy(value) : String(value)] = value;
+    return acc;
+  }, {} as { [key: string]: T });
+
```

Comment:
I'd consider switching to a Map here as [allegedly Maps are more optimized for additions/deletions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) (though we'd have to benchmark).

Reply:
Considering it is scarcely used (currently on `boundElements` only), it would not likely result in any perf. benefits. Actually it might be even worse, since the idea behind `Delta` is that it works on plain array / object, which do not have to be instantiated on load (i.e. when restoring from storage, receiving from other clients, etc.). Also, if we would instantiate Map here for each array, we would have to then go back and forth between the Map and Object, since all the `Delta` utility methods, such as `getLeftDifferences`, `getRightDifferences` work on top of objects.